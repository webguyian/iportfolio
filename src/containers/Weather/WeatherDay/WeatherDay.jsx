import React from 'react';
import PropTypes from 'prop-types';

import Icon from 'components/Icon/Icon';
import Text from 'components/Text/Text';

import { getWeekday, formatTemp } from 'modules/weather/helpers';

const WeatherDay = props => {
  const { icon, time, temperatureHigh, temperatureLow } = props;
  const baseClass = 'weather-day';
  const weekday = getWeekday(time);

  return (
    <li className={baseClass}>
      <Text className={`${baseClass}-name`}>{weekday}</Text>
      <Icon svg className={`${baseClass}-icon`} name={icon} />
      <div className={`${baseClass}-temperature`}>
        <Text>{formatTemp(temperatureHigh, false)}</Text>
        <Text>{formatTemp(temperatureLow, false)}</Text>
      </div>
    </li>
  );
};

WeatherDay.propTypes = {
  icon: PropTypes.string.isRequired,
  temperatureHigh: PropTypes.number,
  temperatureLow: PropTypes.number,
  time: PropTypes.number.isRequired
};

export default WeatherDay;
