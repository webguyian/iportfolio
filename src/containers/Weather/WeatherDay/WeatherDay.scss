.weather-day {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: space-between;
  margin-bottom: $gutter-s;
  text-align: left;

  &-name {
    width: 100px;
  }

  &-icon {
    width: 40px;
    height: 40px;
    fill: $yellow;
  }

  &-temperature {
    > .ui-text--body {
      display: inline-block;
      width: 20px;

      &:last-child {
        margin-left: $gutter;
        color: $gray-light;
      }
    }
  }
}
