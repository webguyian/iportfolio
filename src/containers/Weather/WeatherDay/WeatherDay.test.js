import React from 'react';
import { create } from 'react-test-renderer';

import WeatherDay from './WeatherDay';

describe('<WeatherDay />', () => {
  const props = {
    icon: 'clear-day',
    temperatureHigh: 69.41,
    temperatureLow: 51.43,
    time: 1583726400
  };

  it('renders correctly', () => {
    const component = create(<WeatherDay {...props} />);

    expect(component).toMatchSnapshot();
  });
});
