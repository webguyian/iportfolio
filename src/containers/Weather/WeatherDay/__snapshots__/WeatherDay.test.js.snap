// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherDay /> renders correctly 1`] = `
<li
  className="weather-day"
>
  <span
    className="ui-text ui-text--body weather-day-name"
  >
    Tuesday
  </span>
  <svg
    className="ui-icon ui-icon-clear-day weather-day-icon"
    focusable="false"
    role="presentation"
  >
    <use
      height="100%"
      width="100%"
      xlinkHref="#clear-day"
    />
  </svg>
  <div
    className="weather-day-temperature"
  >
    <span
      className="ui-text ui-text--body"
    >
      69
    </span>
    <span
      className="ui-text ui-text--body"
    >
      51
    </span>
  </div>
</li>
`;
